src/js/components/FilterInputText.js

```diff
@@ -1,93 +1,112 @@
-    };
-  },
-
-  getInitialState: function () {
-    return {
+    this.state = {
       focus: false
     };
-  },
 
-  componentDidUpdate: function () {
-    if (this.state.focus) {
-      this.refs.filterInput.focus();
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.state !== this.state.focus && this.state.focus && this.inputField) {
+      this.inputField.focus();
     }
-  },
+  }
+
+  handleChange(e) {
+    let {target} = e;
 
-  handleChange: function (e) {
-    e.preventDefault();
-    this.props.handleFilterChange(this.refs.filterInput.value,
+    this.props.handleFilterChange(target.value,
       ServiceFilterTypes.TEXT);
-  },
+  }
 
-  handleClearInput: function () {
+  handleClearInput() {
     this.props.handleFilterChange('', ServiceFilterTypes.TEXT);
-  },
+  }
 
-  handleBlur: function () {
+  handleInputChange() { }
+
+  handleBlur() {
     this.setState({
       focus: false
     });
-  },
+  }
 
-  handleFocus: function () {
+  handleFocus() {
     this.setState({
       focus: true
     });
-  },
-
-  renderClearIcon: function (props) {
-
-    if (props.searchString) {
-      let color = 'white';
-
-      if (!props.inverseStyle) {
-        color = 'purple';
-      }
-
-      return (
-        <span className="form-control-group-add-on form-control-group-add-on-append">
-          {props.sideText}
-          <a onClick={this.handleClearInput}>
-          <Icon
-            family="mini"
-            id="ring-close"
-            size="mini"
-            className="clickable"
-            color={color} />
-          </a>
-        </span>
-      );
+  }
+
+  getInputField() {
+    let {inverseStyle, placeholder, searchString} = this.props;
+
+    let inputClasses = classNames({
+      'form-control filter-input-text': true,
+      'form-control-inverse': inverseStyle
+    });
+
+    return (
+      <input
+        className={inputClasses}
+        placeholder={placeholder}
+        onChange={this.handleChange}
+        ref={(ref) => this.inputField = ref}
+        type="text"
+        value={searchString} />
+    );
+  }
+
+  getClearIcon() {
+    let {inverseStyle, sideText} = this.props;
+    let color = 'white';
+
+    if (!inverseStyle) {
+      color = 'purple';
     }
 
-    return null;
-  },
+    return (
+      <span className="form-control-group-add-on form-control-group-add-on-append">
+        {sideText}
+        <a onClick={this.handleClearInput}>
+        <Icon
+          family="mini"
+          id="ring-close"
+          size="mini"
+          className="clickable"
+          color={color} />
+        </a>
+      </span>
+    );
+  }
+
+  renderClearIcon() {
+    if (this.props.searchString) {
+      return this.getClearIcon();
+    }
+  }
 
-  render: function () {
-    var props = this.props;
-    var focus = this.state.focus;
+  render() {
+    let props = this.props;
```

Comment:
Seeing we're making adjustments, lets just pull out the keys we need -> `let {inverseStyle} = this.props;`